blivet/formats/swap.py

```diff
@@ -167,6 +174,14 @@ def _teardown(self, **kwargs):
     def _create(self, **kwargs):
         log_method_call(self, device=self.device,
                         type=self.type, status=self.status)
-        blockdev.swap.mkswap(self.device, label=self.label)
+        if self.new_uuid is None:
+            blockdev.swap.mkswap(self.device, label=self.label)
+        else:
+            if not self.uuid_format_ok(self.new_uuid):
+                raise FSWriteUUIDError("bad UUID format for swap filesystem")
+            extra = [blockdev.ExtraArg("-U", self.new_uuid)]
```

Comment:
You should be able to just pass the ``{"-U": self.uuid}`` dictionary as ``extra`` making things look more pythonic.